tests/test_mad.py

```diff
@@ -0,0 +1,577 @@
+    radius = np.sqrt(x**2 + y**2)       # Matrix values contain *normalised* radius from centre.
+    radius[int(np.round(rows/2+1)),int(np.round(cols/2+1))] = 1 # Get rid of the 0 radius value in the middle
+    radius = np.log(radius+1e-12)
+
+    theta = np.arctan2(-y,x)              # Matrix values contain polar angle.
+    # (note -ve y is used to give +ve
+    # anti-clockwise angles)
+    sintheta = np.sin(theta)
+    costheta = np.cos(theta)
+
+    thetaSigma = math.pi/norient/dThetaOnSigma  # Calculate the standard deviation of the
+
+    logGabors = []
+    for s in range(nscale):                  # For each scale.
+        # Construct the filter - first calculate the radial filter component.
+        fo = 1.0/wavelength[s]                  # Centre frequency of filter.
+        rfo = fo/0.5                         # Normalised radius from centre of frequency plane
+        # corresponding to fo.
+        tmp = -(2 * np.log(sigmaOnf)**2)
+        tmp2= np.log(rfo)
+        logGabors.append(np.exp( (radius-tmp2)**2 /tmp))
+        logGabors[s][int(np.round(rows/2)), int(np.round(cols/2))]=0
+
+
+    E0 = [[],[],[],[]]
+    for o in range(norient):                   # For each orientation.
+        angl = o*math.pi/norient           # Calculate filter angle.
+        
+        ds = sintheta * np.cos(angl) - costheta * np.sin(angl)     # Difference in sine.
+        dc = costheta * np.cos(angl) + sintheta * np.sin(angl)     # Difference in cosine.
+        dtheta = np.abs(np.arctan2(ds,dc))                           # Absolute angular distance.
+        spread = np.exp((-dtheta**2) / (2 * thetaSigma**2))      # Calculate the angular filter component.
+        
+        for s in range(nscale):        # For each scale.
+
+            filter = fftshift(logGabors[s] * spread)
+            filter = torch.from_numpy(filter).reshape(1,1,rows,cols,1).repeat(1,C,1,1,2).to(im.device)
+            # c =  imagefft * filter
+            e0 = torch.ifft( imagefft * filter, 2 )
+            E0[o].append(e0)
+
+    return E0
+        
+def lo_index(ref, dst):
+    gabRef  = gaborconvolve( ref )
+    gabDst  = gaborconvolve( dst )
+    s = [0.5/13.25, 0.75/13.25, 1/13.25, 5/13.25, 6/13.25]
+
+    BSIZE = 16
+    mp = 0
+    for gb_i in range(4):
+        for gb_j in range(5):
+            stdref, skwref, krtref = ical_stat( abs( gabRef[gb_i][gb_j] ) )
+            stddst, skwdst, krtdst  = ical_stat( abs( gabDst[gb_i][gb_j] ) )
+            mp = mp + s[gb_i] * ( torch.abs( stdref - stddst ) + 2*torch.abs( skwref - skwdst ) +  torch.abs( krtref - krtdst ) ) 
+
+    # mp2 = mp[:,:, BSIZE+1:-BSIZE-1, BSIZE+1:-BSIZE-1]
+    B, C, rows, cols = mp.shape
+    return torch.norm( mp.reshape(B,C,-1) , dim=2 ) / np.sqrt(rows * cols) 
+
+
+def mad(x:torch.Tensor, y: torch.Tensor, reduction: str = 'mean',
+            data_range: Union[int, float] = 1., scales: int = 3, subsample: bool = True,
+            c: float = 30.0, alpha: float = 4.2) -> torch.Tensor:
```

Comment:
`c` and `alpha`variables are not used. Add `thresh1`, `thresh2` and others instead to make function more flexible.